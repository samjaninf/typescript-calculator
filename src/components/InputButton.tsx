import { BoundAction } from '../types'
import { Input } from '../redux/modules/operations'
import { StyleAttribute } from 'glamor'
import { h } from 'preact'
import Button from './Button'

type InputButtonProps = {
  action: BoundAction,
  className?: StyleAttribute,
  value: Input,
}

const bindValue = ( action: BoundAction, value: Input ): BoundAction => () => action(value)

const InputButton = ({ action, className, value }: InputButtonProps): JSX.Element => (
  <Button
    action={bindValue(action, value)}
    className={className}
    text={value}
  />
)

export default InputButton
