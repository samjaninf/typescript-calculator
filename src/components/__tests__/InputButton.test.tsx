import InputButton from '../InputButton'
import snapshot from './utils/snapshot'

const action = ( payload: string ) => () => ({ payload, type: 'INPUT' })

describe('InputButton', () => {
  it('renders', () => snapshot(InputButton))
  it.skip('passes action property', () => snapshot(InputButton, { action }))
  it('passes className property', () => snapshot(InputButton, { className: 'css-1j2u492' }))
  it('passes value property', () => snapshot(InputButton, { value: '1' }))
})
