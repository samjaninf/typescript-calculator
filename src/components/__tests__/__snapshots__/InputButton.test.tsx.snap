// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputButton passes className property 1`] = `"<Button class=\\"css-1j2u492\\"></Button>"`;

exports[`InputButton passes value property 1`] = `"<Button text=\\"1\\"></Button>"`;

exports[`InputButton renders 1`] = `"<Button></Button>"`;
